phoenix-core/src/it/java/org/apache/phoenix/end2end/index/MutableIndexIT.java

```diff
@@ -899,6 +899,39 @@ public void testUpdateNonIndexedColumn() throws Exception {
       }
   }
 
+    @Test
+    public void testDeleteCount_nonPK() throws Exception {
+        String schemaName = generateUniqueName();
+        String dataTableName = "TBL_" + generateUniqueName();
```

Comment:
is there a reason we need this test carried over from the immutable index test but not the others?

Reply:
I just wanted to make sure I didn't break anything for mutable indexes so added a test to verify. The bug only applies to immutable index but the code is shared between both immutable and mutable. I could carry all the tests. What do you suggest ?

Reply:
I'd lean toward carrying them all over. It seems like we had a testing gap around affected row counts and you've written a good set of tests to cover that gap for both kinds of table (even if we have no reason to think there's anything wrong with mutable tables). 

With the right common helper methods it won't add much code and I don't expect it to lengthen test suite runtime much.

Reply:
Sounds good